client/landing/stepper/declarative-flow/internals/steps-repository/newsletter-setup/index.tsx

```diff
@@ -91,6 +99,10 @@ const NewsletterSetup: Step = ( { navigation } ) => {
 		}
 	};
 
+	const onPaidSubscribersChanged = ( event: ChangeEvent< HTMLInputElement > ) => {
+		setPaidSubscribers( !! event?.target.checked );
```

Comment:
could we use `[setGoalsOnSite](https://github.com/Automattic/wp-calypso/blob/f784af1dcb5c092376d6fb3f99c7f52ed134c10d/packages/data-stores/src/site/actions.ts#L339)` directly instead?

Reply:
Would be lovely but that would assume the site exists already. The site gets created only later after this step.

Meanwhile we just store the info in state, and pull it from there so that everything works if they go back/forth in the browser or refresh the page or something silly.

They can return to this page from launchpad so we might need to do something like that; follow up PR material.